bundles/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/handler/EvohomeHeatingZoneHandler.java

```diff
@@ -82,16 +89,115 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                 if (EvohomeBindingConstants.ZONE_SET_POINT_CHANNEL.equals(channelId)
                         && command instanceof DecimalType) {
                     double newTemp = ((DecimalType) command).doubleValue();
-                    if (newTemp == CANCEL_SET_POINT_OVERRIDE) {
+                    // Get a local copy of schedule when needed, saves on api calls
+                    DailySchedules schedule = getDailySchedules();
+                    double expectedTemp = getCurrentSetPoint(schedule);
+                    if (newTemp == CANCEL_SET_POINT_OVERRIDE || newTemp == expectedTemp) {
                         bridge.cancelSetPointOverride(getEvohomeThingConfig().id);
+                        updateState(EvohomeBindingConstants.ZONE_SET_POINT_CHANNEL, new DecimalType(expectedTemp));
+                        return;
                     } else if (newTemp < 5) {
                         newTemp = 5;
                     }
                     if (newTemp >= 5 && newTemp <= 35) {
-                        bridge.setPermanentSetPoint(getEvohomeThingConfig().id, newTemp);
+                        int overrideMode = getOverrideMode();
+                        switch (overrideMode) {
+                            case EvohomeBindingConstants.SETPOINT_OVERRIDE_TEMPORARY_SCHEDULE_NEXT:
+                                bridge.setTemporarySetPoint(getEvohomeThingConfig().id, newTemp,
+                                        getNextSetPointChange(schedule));
+                                break;
+                            case EvohomeBindingConstants.SETPOINT_OVERRIDE_TEMPORARY_TIME_ADD:
+                                bridge.setTemporarySetPoint(getEvohomeThingConfig().id, newTemp,
+                                        LocalDateTime.now().plusMinutes(getOverrideTime()));
+                                break;
+                            default:
+                                bridge.setPermanentSetPoint(getEvohomeThingConfig().id, newTemp);
+                                break;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private DailySchedules getDailySchedules() {
+        return this.getEvohomeBridge().getZoneSchedule("temperatureZone", this.getId());
+    }
+
+    private LocalDateTime getNextSetPointChange() {
+        return getNextSetPointChange(this.getDailySchedules());
+    }
+
+    private LocalDateTime getNextSetPointChange(DailySchedules schedule) {
```

Comment:
Would it make sense to add this as a member method of `DailySchedule(s)`? Also, I find it hard to figure out what exactly is going on here. It think you are trying to get the next setpoint based on the given week schedule? Maybe it could use some restructuring to make the code more easy to read.

Reply:
Certainly could be done as a member method. 

And yes, thats exaclty what it does, exactly it gets the time of the setpoint, not the whole thing